navigation/MainNavigationContext.ts

```diff
@@ -0,0 +1,18 @@
+import { createContext } from "react";
+import { Episode } from "../types";
+
+// All screens that exist that do not need props passed to them, if props need to be passed, add it to NavigationState
+export enum NavigationScreens {
+  intro,
+  createEpisode,
+  predictions,
+}
+
+// All possible states that we can be in. Add your navigation state here for new screens
+export type NavigationState =
+  | NavigationScreens
+  | { episodes: Episode[] }
+  | { predictions: String[] };
```

Comment:
It might be a good idea to take advantage of us using typescript here and define each of these as types so you can differentiate between them with having a "type" field instead of by checking if it's an object and then if it has an episodes field when you're using it in MainNavigator. 
Also naming each of the types might make it clearer what they're for, since a list of episodes doesn't make it super clear what state it's in.